### PR TITLE
Exit nonzero if exec fails

### PR DESCRIPTION
**Pull request by chriskuehl**

### CODE REVIEW

**Comment by chriskuehl** [comment]
This fixes #32

File: dumb-init.c [hunk 1]
```diff
@@ -128,6 +128,10 @@ int main(int argc, char *argv[]) {
         }
 
         execvp(argv[1], &argv[1]);
+
+        // if this point is reached, exec failed, so we should exit nonzero
+        DEBUG("execvp failed in child; errno=%d %s\n", errno, strerror(errno));
+        exit(2);
     } else {
```
Comment (bukzor) [inline dumb-init.c]: The user needs more feedback in this case.
This isn't a debugging situation.
For example, bash prints:

```
$ wattttt foo
-bash: wattttt: command not found
```

File: tests/child_processes_test.py [hunk 2]
```diff
@@ -123,3 +123,10 @@ def test_processes_dont_receive_term_on_exit_if_no_setsid(
         assert child_stdout.readline() == str(signum).encode('ascii') + b'\n'
 
     os.kill(child_pid, signal.SIGKILL)
+
+
+def test_fails_nonzero_with_bad_exec(both_debug_modes, both_setsid_modes):
+    """If dumb-init can't exec as requested, it should exit nonzero."""
+    proc = Popen(('dumb-init', '/doesnotexist'))
+    proc.wait()
+    assert proc.returncode != 0
```
Comment (bukzor) [inline tests/child_processes_test.py]: please assert the output

File: dumb-init.c [hunk 3]
```diff
@@ -128,6 +128,16 @@ int main(int argc, char *argv[]) {
         }
 
         execvp(argv[1], &argv[1]);
+
+        // if this point is reached, exec failed, so we should exit nonzero
+        fprintf(
+            stderr,
+            "Failed to launch process '%s' (errno %d: %s)\n",
```
Comment (bukzor) [inline dumb-init.c]: keeping closer to the other unix commands:

'dumb-init: %s: %s', argv[1], strerror(errno)

**Comment by bukzor** [comment]
:shipit:

this is probably a bugfix version bump?

**Pull request merged**